### PR TITLE
Drain all messages on flush to avoid the odd signaling on every message.

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli and @geirst PR

### CODE REVIEW

File: storage/src/vespa/storage/persistence/persistencethread.cpp [hunk 1]
```diff
@@ -950,22 +949,15 @@ PersistenceThread::run(framework::ThreadHandle& thread)
...
-    flushMonitorGuard.broadcast();
 }
 
 void
 PersistenceThread::flush()
 {
-    vespalib::MonitorGuard flushMonitorGuard(_flushMonitor);
-    if (!_closed) {
-        flushMonitorGuard.wait();
+    while (_env._fileStorHandler.getQueueSize() != 0) {
```
Comment (vekterli) [inline storage/src/vespa/storage/persistence/persistencethread.cpp:959]: Could potentially just check the queue size for the thread's stripe, but `getQueueSize()` is a superset of that stripe so shouldn't matter in practice.

**Review by vekterli — APPROVED** [review]
:+1: `flush()` seems to only be called at shutdown, so no point in having extra overhead per message just to be more "efficient" at that

**Review by geirst — APPROVED** [review]
👍

File: storage/src/vespa/storage/persistence/persistencethread.cpp (hunk 1 above)
Comment (baldersheim) [reply storage/src/vespa/storage/persistence/persistencethread.cpp:960]: I will add an api for that after disk concept is removed.

**Pull request merged**